sparse/mlir_backend/_conversions.py

```diff
@@ -103,7 +101,10 @@ def _from_scipy(arr: ScipySparseArray, copy: bool | None = None) -> Array:
                 .build()
             )
 
-            return from_constituent_arrays(format=coo_format, arrays=(pos, row, col, data), shape=arr.shape)
+            ret = from_constituent_arrays(format=coo_format, arrays=(pos, row, col, data), shape=arr.shape)
+            if not copy:
+                _hold_ref(ret, arr)
```

Comment:
These are the critical lines -- we just need to make sure that the SciPy COO array is alive at least as long as the constructed coo array.